lib/http-then.js

```diff
@@ -116,12 +122,15 @@ function getWebSocketStream(options, generator, async = true) {
     });
   });
 
-  return { promise, cancellation: () => cancellation() };
+  return { promise, cancellation: () => cancellation(), sendData: (data) => sendData(data), getSocket:() => getSocket() };
```

Comment:
I would remove the `sendData` attribute as it's redundant with the socket attribute.

Reply:
More generally, I would avoid leaking the socket. It's not that it's a bad idea, but I find it inconsistent with the generator thing.

In the short term, I would add a readable stream option to the `get` method, (and rename the `stream` argument to `generator`), that would be piped to the socket. In our use case, the _exec_ widget would provide that readable stream handle.

In a second step, we may want to replace the generator with a writable stream option to be even more consistent.

Reply:
Let me work on this.

Reply:
Should be done in af1069f21ba6cc635f4f97785c6b1c70f4a1b95f.